contracts/colonyNetwork/ColonyNetworkExtensions.sol

```diff
@@ -0,0 +1,128 @@
+/*
+  This file is part of The Colony Network.
+
+  The Colony Network is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  The Colony Network is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with The Colony Network. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+pragma solidity 0.5.8;
+pragma experimental ABIEncoderV2;
+
+import "../colony/ColonyDataTypes.sol";
+import "../colonyNetwork/IColonyNetwork.sol";
+import "../extensions/ColonyExtension.sol";
+import "./ColonyNetworkStorage.sol";
+
+
+contract ColonyNetworkExtensions is ColonyNetworkStorage {
+
+  // Public functions
+
+  function addExtension(bytes32 _extensionId, address _resolver)
+    public
+    stoppable
+    calledByMetaColony
+  {
+    uint256 version = getResolverVersion(_resolver);
+    require(version > 0, "colony-network-extension-bad-version");
+    resolvers[_extensionId][version] = _resolver;
```

Comment:
This shouldn't be able to be changed, and should probably also check that the previous version is set.

Reply:
I can make this change, but I want to be clear that this actually does nothing from a security perspective, since the upstream Colony resolvers are not similarly locked, so this guarantees nothing. I would advocate for leaving this as-is and changing how we handle resolvers for both extensions and colony versions together. Otherwise, someone reading the code could get a false sense of security. Do you agree?

Reply:
> the upstream Colony resolvers are not similarly locked

I'm not sure what you mean by this? Can you explain?

Do you mean that the extension resolvers themselves are still able to have the functions they have implemented / the particular implementations that they point at changed still? That's only been the case while we've been finding our feet and I've kept those permissions, and won't be the case at some point in the future where we feel comfortable. We'll certainly be comfortable giving up those permissions for extensions sooner than we have been for the colony resolvers themselves.

Even if that were always the case, it is not true to say that my proposed changes guarantee nothing - they give us two useful guarantees to lean on:

A colony that has version x of an extension installed is using the resolver at a particular address.
Existing colonies using the extension are able to upgrade to the new version that has been added.

Reply:
Alright, let's do it.